### ISSUE TITLE
SqlRetryLogic propogation from SqlConnection to SqlCommand

### ISSUE THREAD

**Issue by vakuras** [dotnet/SqlClient#1869]
Currently (AFAIK), creating a new SqlCommand, doesn't copy the SqlRetryLogic configuration from the connection.
Wondering if there is a plan to support this ability?

**Comment by lcheunglci** [comment]
Thanks for bringing this to our attention.  When you [add a SqlRetryLogic](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlconnection.retrylogicprovider?view=sqlclient-dotnet-standard-5.0) in the SqlConnection and creating the `SqlCommand` using that connection and get the `SqlDataReader` from it using [ExecuteReader](https://learn.microsoft.com/en-us/dotnet/api/microsoft.data.sqlclient.sqlcommand.executereader?view=sqlclient-dotnet-standard-5.0#microsoft-data-sqlclient-sqlcommand-executereader) and you can see in the [source](https://github.com/dotnet/SqlClient/blob/b3cfdbbbf9f126f2d0447feb97ae592f19d1b498/src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlCommand.cs#L2416) that it does use the retry logic, however, there is another bug logged for .NET Framework when not directly using the `ExecuteReader` from `SqlDataAdapter` where it references `DbDataAdapter`'s implementation which lacks the retry logic as noted in this issue https://github.com/dotnet/SqlClient/issues/1591 and https://github.com/dotnet/SqlClient/pull/1606 was created to address it; however, due to the fact that .NET Core did not require this fix required some further investigation which is why the PR is still opened.

**Comment by lcheunglci** [comment]
Also, could you provide a repro sample to confirm if it's the same issue as mentioned above?

**Comment by cg110** [comment]
Hi I came across this problem.  If I set sqlcommand.RetryLogicProvider  CreateCommand doesn't hook it up to the commands.  I'm not sure if it should be a flag to buy into the sqlcommands inherit/set it from the connction.

Example code (using .net 8 against 6.0.1 nuget)

```
using Microsoft.Data.SqlClient;

var builder = new SqlConnectionStringBuilder()
{
    DataSource = "localhost",
    InitialCatalog = "master",
    IntegratedSecurity = true,
    TrustServerCertificate = true,
};

var connection = new SqlConnection(builder.ToString());

var retryProvider = SqlConfigurableRetryFactory.CreateExponentialRetryProvider(new() { MaxTimeInterval = TimeSpan.FromSeconds(10), NumberOfTries = 3 }); ;

connection.RetryLogicProvider = retryProvider;
connection.Open();

var cmd = connection.CreateCommand();
cmd.CommandText = "SELECT @@VERSION";

var result = cmd.ExecuteReader();
result.Read();

Console.WriteLine("SQL Server version: {0}, retry is using the cmd one: {1}", result.GetString(0), cmd.RetryLogicProvider == retryProvider);
```
I would expect the cmd.RetryLogicProvider to be the one I set on the command, but it's the default "none" provider.

**Comment by samsharma2700** [comment]
Yes, this is by design. The retry‑logic feature in` Microsoft.Data.SqlClient` is opt‑in on each object, and creating a `SqlCommand `via `SqlConnection.CreateCommand()` does not automatically inherit the connection’s `RetryLogicProvider`. You have to explicitly set it on the command `cmd.RetryLogicProvider = retryProvider;`. I do not think this is on our roadmap for now. If needed please open a feature request.
